wp-graphql-meta.php

```diff
@@ -22,8 +22,15 @@
 		array( 'user' )
 	);
 
+	// Some WPGraphQL type names differ from their post types.
+	$mapping = array(
+		'attachment' => 'mediaItem',
+	);
+
 	foreach ( $all_types as $type ) {
-		add_filter( "graphql_{$type}_fields", function ( $fields ) use ( $type ) {
+		$graphql_type = isset( $mapping[ $type ] ) ? $mapping[ $type ] : $type;
```

Comment:
I would recommend getting the `graphql_single_name` from the `post_type_object` instead of relying on a static map. 

It is possible for folks to edit the post_type_object, so there's no guarantee that the type for `attachments` will always be `mediaItem`, the only guarantee is that the type in the Schema will be the `post_type_object()->graphql_single_name`

So, `$graphql_type = ! empty( get_post_type_object( $type )->graphql_single_name ) ? get_post_type_object( $type )->graphql_single_name : $type;` might work, or something to that tune?

Reply:
just an example, if I wanted the Schema for attachments to be `attachments` instead of `mediaItems` I could do something to this tune, and the schema wouldn't have `mediaItem` or `mediaItems` in it at all, essentially breaking this mapping: 

```
add_filter( 'register_post_type_args', function( $args, $post_type ) {
  if ( 'attachment' === $post_type ) {
      $args['graphql_single_name'] = 'attachment';
      $args['graphql_plural_name'] = 'attachments';
  }
}, 20, 2);
```

Reply:
makes sense, will push something

Reply:
hmmm, `show_in_graphql` / `graphql_single_name` aren't set until `do_graphql_request`, whereas this plugin is running on `graphql_init`. seems like one or the other should move, but not sure which

Reply:
It appears that `show_in_graphql`, `graphql_single_name`, `graphql_plural_name` can be replaced by `show_in_rest` and `get_post_type_labels()` and `get_taxonomy_labels()` for posts and taxonomies labels.